Add a new "direct" biblio display style

brycelelbach opened the pull request:
Add a new biblio display style, "direct", which displays references by their shortname and links to the referenced document. Closes #2032.

tabatkins commented on bikeshed/shorthands/oldShorthands.py:
```diff
@@ -380,10 +380,17 @@ def biblioReplacer(match):
-    elif displayInline or displayIndex:
-        attrs["data-biblio-display"] = "inline" if displayInline else "index"
+    displayDirect = "direct" in modifiers
+    if (displayInline + displayIndex + displayDirect) > 1:
```
I hate that Python bool is technically a subclass of int, but I suppose it has its uses...

tabatkins commented on docs/index.bs:
```diff
@@ -1070,10 +1070,11 @@ There are several additional optional keys:
-		or displaying as their title and linking straight to the referenced document.
+		displaying as their shortname and linking to the bibliography index ("index"),
+		displaying as their title and linking straight to the referenced document ("inline"), or
+    displaying as their shortname and linking to the referenced document ("direct").
```
your indent is broken

tabatkins commented on docs/index.bs:
```diff
@@ -2928,13 +2929,14 @@ after the biblio name.
 * By default, biblio autolinks are formatted like you typed, just with a single set of square brackets:
 	`[[FOO]]` will produce a link with the text "\[FOO]",
 	which links to the bibliography.
+  You can make them display "direct", where they link straight to the document,
```
broken indent

tabatkins reviewed (changes requested):
The feature looks good. Only problem is the busted indents in the docs, which'll break the formatting when I actually process them.

When you fix those, would you mind running Bikeshed over it and including the results in the PR, too?

brycelelbach commented:
D'oh. Looks like I missed the fact that the docs use tabs! I was wondering if I should regenerate them.

I've fixed the indents and added regenerated docs to the PR. I also rebased on `main` (and regenerated the tests and docs afterwards just to be safe).

brycelelbach commented:
Thanks!
